src/tscore/HostLookup.cc

```diff
@@ -428,9 +428,13 @@ CharIndex::iterator::advance() -> self_type &
       break;
     } else if (state.block->array[state.index].block != nullptr) {
       // There is a lower level block to iterate over, store our current state and descend
-      q[cur_level++] = state;
-      state.block    = state.block->array[state.index].block.get();
-      state.index    = 0;
+      if (static_cast<int>(q.size()) <= cur_level) {
+        q.push_back(state);
+      } else {
+        q[cur_level] = state;
```

Comment:
the old code incremented `cur_level` after the lookup, but the new one doesn't, is that on purpose?

Reply:
thanks for that! Forgot to bring that over.

Reply:
Hmmm, is this still missing the cur_level++ ?

Reply:
no, it's just below this block (line 437)